backend/battle/tests/test_view.py

```diff
@@ -74,3 +81,178 @@ def test_verify_if_not_returns_winner_missing_pokemons_teams(self):
+
+    def test_data_returns_a_lot_battle_ids(self):
+        self.client.login(username=self.user.email, password='admin')
+        battles = baker.make('battle.Battle', creator=self.user, _quantity=100)
+        response = self.client.get(reverse('battle_list'))
+        response_qs = response.context_data.get('battle_list')
+        self.assertCountEqual(battles, response_qs)
+
+    def test_data_an_error_when_the_user_is_not_logged(self):
+        response = self.client.get(reverse('battle_list'))
+        self.assertEqual(response.status_code, 302)
+
+    def test_data_diff_to_verify_it_returns_exactly_len_of_battle_list(self):
+        self.client.login(username=self.user.email, password='admin')
+
+        baker.make('battle.Battle', creator=self.user, _quantity=10)
+        battles_initial = Battle.objects.filter(creator=self.user)
+        response_initial = self.client.get(reverse('battle_list'))
+        response_qs_initial = response_initial.context_data.get('battle_list')
+
+        self.assertCountEqual(battles_initial, response_qs_initial)
+
+        baker.make('battle.Battle', creator=self.user, _quantity=90)
+        battles_updated = Battle.objects.filter(creator=self.user)
+        response_updated = self.client.get(reverse('battle_list'))
+        response_qs_updated = response_updated.context_data.get('battle_list')
+
+        self.assertCountEqual(battles_updated, response_qs_updated)
+
+
+class BattleCreateViewTest(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.client = Client()
+        self.creator = User.objects.create(email='example@example.com')
+        self.creator.set_password('admin')
+        self.creator.save()
+        self.opponent = baker.make('users.User')
+
+    def test_create_battle_successfully(self):
+        battle_data = {
+            "creator": self.creator.id,
+            "opponent": self.opponent.email,
+        }
+        self.client.login(username=self.creator.email, password='admin')
+        self.client.post(reverse('battle'), battle_data)
+        battle = Battle.objects.filter(creator=self.creator, opponent=self.opponent)
+        self.assertTrue(battle)
+
+    def test_if_returns_error_when_creator_and_opponent_are_the_same_user(self):
+        battle_data = {
+            "creator": self.creator.id,
+            "opponent": self.creator.email,
+        }
+        self.client.login(username=self.creator.email, password='admin')
+        response = self.client.post(reverse('battle'), battle_data)
+        self.assertEqual(
+            response.context_data['form'].errors['__all__'][0],
+            "ERROR: You can't challenge yourself.")
+
+    def test_if_returns_error_when_opponent_doesnt_exist(self):
+        battle_data = {
+            "creator": self.creator.id,
+            "opponent": 100,
+        }
+        self.client.login(username=self.creator.email, password='admin')
+        with self.assertRaises(KeyError):
```

Comment:
I think I already commented on this, but we should not let views raises exceptions since it hurts the user experience, we should change to return a validation error

Reply:
yup